assets/app/views/browse/edit-build-config.html

```diff
@@ -0,0 +1,257 @@
+<div class="content">
+  <project-page>
+    <breadcrumbs breadcrumbs="breadcrumbs"></breadcrumbs>
+    <alerts alerts="alerts"></alerts>
+    <div ng-if="!loaded">Loading...</div>
+    <div class="tile" ng-if="buildConfig">
+
+      <div>
+        <h1>
+          Edit {{buildConfig.metadata.name}} Build Config
+        </h1>
+      </div>
+
+          <form class="edit-form" novalidate name="form">
+            <div class="resource-details">
+              <div class="row">
+                <div class="col-lg-6">
+                  <div class="section">
+                    <h3>Configuration</h3>
+                      <dl class="dl-horizontal left">
+                        <div class="form-group">
+                          <div ng-if="updatedBuildConfig.spec.source.type == 'Git'">
+                            <label for="sourceUrl">Git Repository URL</label>
+                            <div>
+                              <!-- Unfortunately, we can't set type="url" because some valid values don't pass browser validation. -->
+                              <input class="form-control"
+                                id="sourceUrl"
+                                name="sourceUrl"
+                                ng-model="updatedBuildConfig.spec.source.git.uri"
+                                type="text"
+                                aria-describedby="from_source_help">
+                            </div>
+                          </div>
+                          <div>
+                            <span class="text-warning" ng-if="form.sourceUrl.$dirty && !sourceURLPattern.test(updatedBuildConfig.sourceUrl)">Git repository should be a URL.</span>
+                          </div>
+                        </div>
+
+                        <div class="form-group">
+                          <div ng-if="updatedBuildConfig.spec.source.type == 'Git'">
+                            <label for="sourceRef">Git Repository Ref</label>
+                            <div>
+                              <!-- Unfortunately, we can't set type="url" because some valid values don't pass browser validation. -->
+                              <input class="form-control"
+                                id="sourceRef"
+                                name="sourceRef"
+                                type="text"
+                                ng-model="updatedBuildConfig.spec.source.git.ref"
+                                aria-describedby="from_source_help"
+                                placeholder="{{updatedBuildConfig.spec.source.git.ref === undefined ? 'master' : updatedBuildConfig.spec.source.git.ref}}">
```

Comment:
We shouldn't set the placeholder text to an actual value. Let's always use "master".